src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/JsonConverters/CVImageBinaryDataJsonConverter.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using RegressionGames.StateRecorder.BotSegments.Models;
+using RegressionGames.StateRecorder.BotSegments.Models.CVSerice;
+
+namespace RegressionGames.StateRecorder.BotSegments.JsonConverters
+{
+    public sealed class CVImageBinaryDataJsonConverter : JsonConverter
+    {
+        public override bool CanConvert(Type objectType)
+        {
+            return typeof(CVImageBinaryData).IsAssignableFrom(objectType);
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            JObject jObject = JObject.Load(reader);
+            // ReSharper disable once UseObjectOrCollectionInitializer - easier to debug when on separate lines
+            CVImageBinaryData actionModel = new();
+            actionModel.data = jObject.GetValue("data").ToObject<byte[]>(serializer);
```

Comment:
(non-blocking, random thought) I forget if we had this idea during the original conversation, but what about an option for pointing this to a relative path to some image that gets loaded on the fly before we make the API call? I know that we have some ideas around showing stuff in some UI for this, but this could have made early debugging easier maybe? Not saying to change it though, just thinking out loud.

Reply:
I won't be changing this for now.

I can see plusses and minuses for both approaches.
Current approach makes each segment file self contained.
JPG file path approach makes it easy to reference /update the same image across many segment files, but makes the files slightly less portable between system (not non-portable, just pathing must be carefully handled by users)

Feel free to make a follow up task for this, but I would suggest we only prioritize it after all the segment types and structure are in and we can play with the tooling a lot and get a feel for it.  That said, I don't think we'll need to wait for external feedback, our internal testing should make this one pretty obvious once we're re-using and stitching segments together.  I'm guessing that this suggestion will only become really needed if the target image changes frequently.

Reply:
@vontell this is holding up merge and blocking my follow up PRs/tasks, please check your outstanding review requests as soon as you can